src/Microsoft.Data/PrimitiveColumn.cs

```diff
@@ -0,0 +1,110 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Data
+{
+    /// <summary>
+    /// A column to hold primitive values such as int, float etc. Other value types are not really supported
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    public partial class PrimitiveColumn<T> : BaseColumn
+        where T : struct
```

Comment:
Do you think we can use `where T : unmanaged` instead?

[https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/where-generic-type-constraint](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/where-generic-type-constraint)

> The unmanaged constraint limits the type parameter to types known as unmanaged types. An unmanaged type is a type that isn't a reference type and doesn't contain reference type fields at any level of nesting. The unmanaged constraint makes it easier to write low-level interop code in C#. This constraint enables reusable routines across all unmanaged types.